core/src/main/groovy/noe/workspace/WorkspaceMultipleTomcats.groovy

```diff
@@ -16,16 +16,16 @@ class WorkspaceMultipleTomcats extends WorkspaceAbstract{
   }
 
   void createAdditionalTomcatsWithRegistrations(int numberOfAdditionalTomcats) {
-    for (int i = 2; i < numberOfAdditionalTomcats + 2; i++) {
-      // We start tests always with only one tomcat
+    int initialNumberOfTomcats = serverController.numberOfTomcatServers()
+    for (int i = initialNumberOfTomcats; i <= (initialNumberOfTomcats + numberOfAdditionalTomcats); i++) {
       String tomcatVersion = TomcatProperties.TOMCAT_MAJOR_VERSION
       String id = "tomcat-$tomcatVersion-${i}"
       log.info("Creating new tomcat server instance: ${id}")
       // if node2 is not defined - create default
       if (!serverController.getTomcatServerIds([TomcatProperties.TOMCAT_MAJOR_VERSION]).contains(id)) {
         String tomcatDir = id
         Tomcat nextServer = (Tomcat) Tomcat.getInstance(basedir, tomcatVersion.toString(), tomcatDir, context)
-        nextServer.createNewServerInstance(id, DefaultProperties.DEFAULT_SHIFT_PORT_OFFSET * (i - 1))
+        nextServer.createNewServerInstance(id, DefaultProperties.DEFAULT_SHIFT_PORT_OFFSET * (i + 1))
```

Comment:
if i == 0 then we'll get -1 * DEFAULT_SHIFT_PORT_OFFSET and that will result in a negative port number.
if i == 1 the we'll get 0 * DEFAULT_SHIFT_PORT_OFFSET and that will be port 0
i + 1 avoids the scenario

Reply:
There shouldn't be a scenario when i = 0 as we always have at least one, right?
And if we do i = 1, you have offset of 2 * DEFAULT_SHIFT_PORT_OFFSET and the server numbering starts at 1.

Reply:
if we +1 like you suggest in the initialization, I agree, but that gets into the second scenario I indicated: 
where i = 1, would mean (1 - 1)  * DEFAULT_SHIFT_PORT_OFFSET will give a port number of 0.

We could elevate it by having another counter j initialized to 1 and then incremented on each loop, that would also remove the need to have +1 or -1

thoughts?

Reply:
The DEFAULT_SHIFT_PORT_OFFSET is an offset, right?
So the default server(tomcat-10-1) has offset (1-1)*101, so no offset.
The next one(tomcat-10-2) has offset (2-1)*101, so 101 offset.
And so on...
Or is my reasoning wrong?
Though my premise depends on what ports the tomcat-10-1 uses. Since it isn't created by this method.